pandas/core/apply.py

```diff
@@ -327,12 +327,16 @@ def agg_list_like(self) -> DataFrame | Series:
         else:
             context_manager = nullcontext()
         with context_manager:
+
+            if self.args:
+                self.args = (self.axis,) + self.args
```

Comment:
I don't think we should modify the state here (e.g. agg_list_like could be called twice on the same object). In addition, when reading this code I think it is expected that `self.args` is just the arguments that the user passed in.

Reply:
Yes, I agree that we should only pass what the user passed in. But this code I originally used to solve the incorrect axis.

If without this code, when user call `df.agg([foo1, foo2], 0, 3, c=4)`, for `arg = [foo1, foo2]`, `self.axis=0, self.args=(3,), self.kwargs={'c':4}`, looks good. But call `new_res = colg.aggregate(arg, *self.args, **self.kwargs)` will create a new class, and `self.axis` will try to set as 3, raise an error.

<img width="581" alt="image" src="https://user-images.githubusercontent.com/77875500/216512277-eeb9d5cd-2044-4e8a-977b-0064d2c7fe87.png">

I add this code for avoid this problem. If we don't avoid the error here, should we raise it directly, or modify the code elsewhere to avoid it?

Reply:
Does passing `self.axis, *self.args, **self.kwargs` to aggregate not work?

Reply:
I have tried this and it causes some other test cases to fail.

Reply:
Indeed, sorry to waste your time with such a question.

Reply:
Or could we 
```python
if self.args:
    new_res = colg.aggregate(arg, self.axis, *self.args, **self.kwargs)
else: 
    new_res = colg.aggregate(arg)
```

pass `self.axis` for every situation is not work fine.

Reply:
I see - makes sense. We need to determine whether `colg.aggregate` has an axis argument or not depending on the type of object that `colg` is. It looks like this axis argument is only on Series / DataFrame.